dkron/queries.go

```diff
@@ -197,5 +198,12 @@ func (a *AgentCommand) queryRPCConfig() ([]byte, error) {
 		"query": QueryRPCConfig,
 	}).Debug("proc: Done receiving acks and responses")
 
+	if rpcAddr == nil {
+		log.WithFields(logrus.Fields{
+			"query": QueryRPCConfig,
+		}).Debug("proc: Failed to receive a config response, retrying")
+		goto queryConfigStart
```

Comment:
This is a pretty hardcore looping, I prefer to add a retry limit control here then fataling, otherwise agents could end up in an endless loop trying to contact inexistent servers.

Reply:
Hi,
Just wonder do we have any fix on this error?
I got into this error ("dial tcp: missing address") quite frequent: around 1 time/1hour in average.
Cannot apply dkron to production with this error :( ?

Reply:
I agree that this could end up in an infinite loop. I'm concerned of the case of the `concurrency: forbid`job though. If there is a limited number of retries, and we stop trying to record that a job finished and the job will never run until the previous run is complete, we just stop executing a job. Maybe there are some alternatives that could be implemented like a timeout parameter on a job? This could give the best of both worlds.

Reply:
The problem is not that the job doesn't finish, it does, but it is unable to communicate with a server. Under my production environment, this doesn't happen, so there should be some subtle network related issues that prevent this query to receive a response in time.

A timeout for waiting for a job execution is one possibility, but you will never know if the job never finished or it finished but you didn't get the response.

We can try several things here, but I will add a timeout here to unblock the PR.

Reply:
@sysadmind @giaplv57 I'm unable reproduce the error, could you try to increase here https://github.com/victorcoder/dkron/blob/master/dkron/agent.go#L222 adding:

```
serfConfig.QueryTimeoutMult = (something > 16)
```

And see if it solves the problem?

Send me an email if you can't build dkron yourself.

Thanks

Reply:
I tried this and set the timeout to 30. It didn't make a difference and within a few minutes I had jobs that failed to record their results with log messages about failing to dial a null address. 

I have a theory that it involves the master dropping the query, but I haven't had time to day to troubleshoot. I will see if I can create more steps to reproduce the exact issue.